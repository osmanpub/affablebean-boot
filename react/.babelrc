 
{
  "presets": ["@babel/preset-react"],
  "env": {
    "test": {
      "plugins": [
        ["istanbul", {
          "exclude": [
            "**/*.spec.js"
          ]
        }]
      ]
    }
  },
  "plugins": ["transform-class-properties", "istanbul"]
}
