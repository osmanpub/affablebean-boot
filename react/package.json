{
  "name": "affablebean",
  "version": "0.1.0",
  "private": true,
  "eslintConfig": {
    "extends": "react-app",
    "plugins": [
      "react-hooks"
    ],
    "rules": {
      "react-hooks/rules-of-hooks": "error",
      "react-hooks/exhaustive-deps": "warn"
    }
  },
  "scripts": {
    "build": "react-scripts build",
    "coveralls": "cat coverage/lcov.info | coveralls",
    "cypress:open": "cypress open",
    "cypress:run": "cypress run",
    "cypress:verify": "cypress verify",
    "eject": "react-scripts eject",
    "report:coverage": "nyc report --reporter=html",
    "report:coverage:text": "nyc report --reporter=text",
    "start": "node_modules/react-scripts/bin/react-scripts.js start &",
    "start:java": "cd .. && sh server && cd react && node_modules/react-scripts/bin/react-scripts.js start &",
    "test": "react-scripts test"
  },
  "browserslist": {
    "production": [
      ">0.2%",
      "not dead",
      "not op_mini all"
    ],
    "development": [
      "last 1 chrome version",
      "last 1 firefox version",
      "last 1 safari version"
    ]
  },
  "dependencies": {
    "@reduxjs/toolkit": "^1.2.5",
    "axios": "^0.19.2",
    "node-rest-client": "^3.1.0",
    "react": "^16.12.0",
    "react-alert": "^6.0.0",
    "react-alert-template-basic": "^1.0.0",
    "react-dom": "^16.12.0",
    "react-hook-form": "^4.9.7",
    "react-redux": "^7.2.0",
    "react-router-dom": "^5.1.2",
    "react-scripts": "3.4.0",
    "redux": "^4.0.5",
    "styled-components": "^5.0.1"
  },
  "devDependencies": {
    "@babel/core": "^7.8.4",
    "@babel/preset-react": "^7.8.3",
    "@cypress/code-coverage": "^1.12.2",
    "@testing-library/jest-dom": "^4.2.4",
    "@testing-library/react": "^9.4.0",
    "@testing-library/user-event": "^7.2.1",
    "@types/axios": "^0.14.0",
    "@types/cypress": "^1.1.3",
    "@types/jest": "^25.1.3",
    "@types/react": "^16.9.21",
    "@types/react-alert": "^5.2.0",
    "@types/react-bootstrap": "^0.32.20",
    "@types/react-dom": "^16.9.5",
    "@types/react-redux": "^7.1.7",
    "@types/react-router-dom": "^5.1.3",
    "@types/react-test-renderer": "^16.9.2",
    "@types/redux": "^3.6.0",
    "@types/redux-mock-store": "^1.0.2",
    "@types/styled-components": "^5.0.0",
    "babel-plugin-istanbul": "^6.0.0",
    "babel-plugin-transform-class-properties": "^6.24.1",
    "coveralls": "^3.0.9",
    "cross-env": "^7.0.0",
    "cypress": "^4.0.2",
    "istanbul-lib-coverage": "^3.0.0",
    "nyc": "^15.0.0",
    "react-test-renderer": "^16.12.0",
    "redux-mock-store": "^1.5.4",
    "typescript": "^3.8.2"
  },
  "nyc": {
    "sourceMap": false,
    "instrument": false
  }
}
