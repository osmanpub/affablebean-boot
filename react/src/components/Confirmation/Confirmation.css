.confirmationPriceColumn {
  width: 90px;
  font-weight: normal;
}

.summaryColumn {
  margin-top: 15px;
  width: 50%;
  float: left;
}
