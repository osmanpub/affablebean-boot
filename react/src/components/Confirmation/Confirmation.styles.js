import styled from "styled-components";

export const ConfirmationText = styled.p`
  margin: 0 20px 20px;
  padding: 10px;
  background-color: #f5eabe;
  float: left;
  border-radius: 4px;
`;
