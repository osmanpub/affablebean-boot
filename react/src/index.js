import React from "react";
import { render } from "react-dom";
import { configureStore } from "redux-starter-kit";
import { Provider } from "react-redux";
import { BrowserRouter as Router, Route } from "react-router-dom";
import reducer from "./reducers";
import App from "./containers/App";
import CategoryProducts from "./containers/CategoryProducts";
import * as serviceWorker from "./serviceWorker";

const store = configureStore({
  reducer
});

render(
  // <React.StrictMode>
  <Provider store={store}>
    <Router>
      <Route path="/" exact component={App} />
      <Route path="/category/:id" component={CategoryProducts} />
    </Router>
  </Provider>,
  // </React.StrictMode>,
  document.getElementById("root")
);

// If you want your app to work offline and load faster, you can change
// unregister() to register() below. Note this comes with some pitfalls.
// Learn more about service workers: http://bit.ly/CRA-PWA
serviceWorker.unregister();
