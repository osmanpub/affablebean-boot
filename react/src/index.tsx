import { configureStore } from "@reduxjs/toolkit";
import React, { lazy, Suspense } from "react";
import { render } from "react-dom";
import { Provider } from "react-redux";
import { BrowserRouter as Router, Route, Switch } from "react-router-dom";
import ErrorBoundary from "./components/ErrorBoundary";
import rootReducer from "./redux";
import * as serviceWorker from "./serviceWorker";
const Cart = lazy(() => import("./containers/Cart"));
const CategoryProducts = lazy(() => import("./containers/CategoryProducts"));
const Checkout = lazy(() => import("./containers/Checkout"));
const Contact = lazy(() => import("./containers/Contact"));
const Home = lazy(() => import("./containers/Home"));
const Privacy = lazy(() => import("./containers/Privacy"));

const store = configureStore({
  reducer: rootReducer
});

render(
  <React.StrictMode>
    <ErrorBoundary>
      <Provider store={store}>
        <Router>
          <Suspense fallback={<div>Loading...</div>}>
            <Switch>
              <Route path="/" exact component={Home} />
              <Route path="/category/:id" component={CategoryProducts} />
              <Route path="/checkout" component={Checkout} />
              <Route path="/contact" component={Contact} />
              <Route path="/privacy" component={Privacy} />
              <Route path="/viewCart/:clear" component={Cart} />
            </Switch>
          </Suspense>
        </Router>
      </Provider>
    </ErrorBoundary>
  </React.StrictMode>,
  document.getElementById("root")
);

// If you want your app to work offline and load faster, you can change
// unregister() to register() below. Note this comes with some pitfalls.
// Learn more about service workers: http://bit.ly/CRA-PWA
serviceWorker.unregister();
