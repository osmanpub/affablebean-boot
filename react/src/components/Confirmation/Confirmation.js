import React from "react";
import { Link } from "react-router-dom";
import { ConfirmationText } from "./Confirmation.styles";
import "./Confirmation.css";

export function Confirmation() {
  const { customer, orderedProducts, orderRecord, products } = this.props.order;

  return (
    <div className="singleColumn">
      <ConfirmationText>
        <strong>
          Your order has been successfully processed and will be delivered
          within 24 hours.
        </strong>
        <br />
        <br />
        Please keep a note of your confirmation number:
        <strong>{orderRecord.confirmationNumber}</strong>
        <br />
        If you have a query concerning your order, feel free to
        <Link to="/contact">contact us</Link>
        <a href="/contact">contact us</a>.
        <br />
        <br />
        Thank you for shopping at the Affable Bean Green Grocer!
      </ConfirmationText>
      <br />
    </div>
  );
}
