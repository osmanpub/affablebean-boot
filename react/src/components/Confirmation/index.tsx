import React from "react";
import { Link } from "react-router-dom";
import { getId } from "../../helpers/utils";
import { Order, OrderedProduct } from "../../interfaces/purchase";
import "./Confirmation.css";
import {
  ConfirmationText,
  DateProcessedRow,
  DeliveryAddressTable,
  DeliveryAddressTableTd,
  DeliverySurchargeCellLeft,
  DeliverySurchargeCellRight,
  OrderSummaryTable,
  TotalCellLeft,
  TotalCellRight
} from "./Confirmation.styles";

type Props = {
  order: Order;
};

export default function Confirmation(props: Props) {
  // BUG:for node, the orderedProducts and products arrays are not working!
  const { customer, orderedProducts, orderRecord, products } = props.order;

  const orderedProductsList = orderedProducts.map(
    (product: OrderedProduct, index: number) => {
      const rowCol = index % 2 === 0 ? "white" : "lightBlue";
      const qty = product.quantity;
      const subtotal = (products[index].price * qty).toFixed(2);

      return (
        <tr className={`${rowCol}`} key={getId(products[index])}>
          <td>{products[index].name}</td>
          <td className="quantityColumn">{qty}</td>
          <td className="confirmationPriceColumn">&euro; {subtotal}</td>
        </tr>
      );
    }
  );

  const tdStyle = {
    padding: "0 20px"
  };

  return (
    <div className="singleColumn">
      <ConfirmationText>
        <strong data-cy="confirm-success">
          Your order has been successfully processed and will be delivered
          within 24 hours.
        </strong>
        <br />
        <br />
        Please keep a note of your confirmation number:
        <strong>{orderRecord.confirmationNumber}</strong>
        <br />
        If you have a query concerning your order, feel free to&nbsp;
        <Link to="/contact">contact us</Link>
        <br />
        <br />
        Thank you for shopping at the Affable Bean Green Grocer!
      </ConfirmationText>
      <div className="summaryColumn">
        <OrderSummaryTable className="detailsTable">
          <tbody>
            <tr className="header">
              <th colSpan={3}>order summary</th>
            </tr>
            <tr className="tableHeading">
              <td>product</td>
              <td>quantity</td>
              <td>price</td>
            </tr>
            {orderedProductsList}
            <tr className="lightBlue">
              <td colSpan={3} style={tdStyle}>
                <hr />
              </td>
            </tr>
            <tr className="lightBlue">
              <DeliverySurchargeCellLeft colSpan={2}>
                <strong>surcharge:</strong>
              </DeliverySurchargeCellLeft>
              <DeliverySurchargeCellRight>
                &euro; {surcharge.toFixed(2)}
              </DeliverySurchargeCellRight>
            </tr>
            <tr className="lightBlue">
              <TotalCellLeft colSpan={2}>
                <strong>total:</strong>
              </TotalCellLeft>
              <TotalCellRight>
                &euro; {orderRecord.amount.toFixed(2)}
              </TotalCellRight>
            </tr>
            <tr className="lightBlue">
              <td colSpan={3} style={tdStyle}>
                <hr />
              </td>
            </tr>
            <tr className="lightBlue">
              <DateProcessedRow colSpan={3}>
                <strong>date processed: </strong>
                <span>{orderRecord.dateCreated}</span>
              </DateProcessedRow>
            </tr>
          </tbody>
        </OrderSummaryTable>
      </div>
      <div className="summaryColumn">
        <DeliveryAddressTable className="detailsTable">
          <tbody>
            <tr className="header">
              <th colSpan={3}>delivery address</th>
            </tr>
            <tr>
              <DeliveryAddressTableTd colSpan={3} className="lightBlue">
                {customer.name}
                <br />
                {customer.address}
                <br />
                {customer.cityRegion}
                <br />
                <hr />
                {customer.email}
                <br />
                {customer.phone}
              </DeliveryAddressTableTd>
            </tr>
          </tbody>
        </DeliveryAddressTable>
      </div>
    </div>
  );
}

const surcharge = 3;
