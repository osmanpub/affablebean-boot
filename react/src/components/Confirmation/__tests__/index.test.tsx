import { render } from "@testing-library/react";
import React from "react";
import { BrowserRouter as Router } from "react-router-dom";
import renderer from "react-test-renderer";
import Confirmation from "..";

const order = {
  customer: {
    id: 1,
    address: "none of your business",
    ccNumber: "1234567890123456",
    cityRegion: "NY",
    email: "joe@bloggs.com",
    name: "joe bloggs",
    phone: "1234567890"
  },
  orderedProducts: [
    {
      quantity: 1
    }
  ],
  orderRecord: {
    id: 1,
    amount: 5.39,
    confirmationNumber: "995604757",
    dateCreated: "1582123900751"
  },
  products: [
    {
      id: 12,
      description: "contain peanuts<br>(3 cookies)",
      name: "chocolate cookies",
      price: 2.39,
      category: { id: 3, name: "bakery" }
    }
  ]
};

const confirmation = (
  <Router>
    <Confirmation order={order} />
  </Router>
);

describe("<Confirmation />", () => {
  it("renders correctly", () => {
    const component = renderer.create(confirmation).toJSON();
    // @ts-ignore
    expect(component).toMatchSnapshot();
  });

  it("show success paragraph", () => {
    const { getByTestId } = render(confirmation);
    const success = getByTestId(/confirm-success/i);
    // @ts-ignore
    expect(success).toBeInTheDocument();
  });

  it("surcharge is 3.00", () => {
    const { getByTestId } = render(confirmation);
    const total = getByTestId(/surcharge/i);
    // @ts-ignore
    expect(total).toHaveTextContent(/3.00/i);
  });

  it("total is 5.39", () => {
    const { getByTestId } = render(confirmation);
    const surcharge = getByTestId(/total/i);
    // @ts-ignore
    expect(surcharge).toHaveTextContent(/5.39/i);
  });

  it("customer name is 'joe bloggs'", () => {
    const { getByTestId } = render(confirmation);
    const name = getByTestId(/name/i);
    // @ts-ignore
    expect(name).toHaveTextContent(/joe bloggs/i);
  });

  it("customer address is 'none of your business'", () => {
    const { getByTestId } = render(confirmation);
    const address = getByTestId(/address/i);
    // @ts-ignore
    expect(address).toHaveTextContent(/none of your business/i);
  });

  it("customer phone is '1234567890'", () => {
    const { getByTestId } = render(confirmation);
    const phone = getByTestId(/phone/i);
    // @ts-ignore
    expect(phone).toHaveTextContent(/1234567890/i);
  });

  it("customer email is 'joe@bloggs.com'", () => {
    const { getByTestId } = render(confirmation);
    const email = getByTestId(/email/i);
    // @ts-ignore
    expect(email).toHaveTextContent(/joe@bloggs.com/i);
  });

  it("customer cityRegion is 'joe bloggs'", () => {
    const { getByTestId } = render(confirmation);
    const region = getByTestId(/cityRegion/i);
    // @ts-ignore
    expect(region).toHaveTextContent(/NY/);
  });
});
