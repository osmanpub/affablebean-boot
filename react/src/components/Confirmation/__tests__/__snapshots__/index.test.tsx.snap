// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Confirmation /> renders correctly 1`] = `
<div
  className="singleColumn"
>
  <p
    className="sc-fznxsB LMNBB"
  >
    <strong
      data-cy="confirm-success"
    >
      Your order has been successfully processed and will be delivered within 24 hours.
    </strong>
    <br />
    <br />
    Please keep a note of your confirmation number:
    <strong>
      995604757
    </strong>
    <br />
    If you have a query concerning your order, feel free to 
    <a
      href="/contact"
      onClick={[Function]}
    >
      contact us
    </a>
    <br />
    <br />
    Thank you for shopping at the Affable Bean Green Grocer!
  </p>
  <div
    className="summaryColumn"
  >
    <table
      className="sc-fzoyTs jLIpLc detailsTable"
    >
      <tbody>
        <tr
          className="header"
        >
          <th
            colSpan={3}
          >
            order summary
          </th>
        </tr>
        <tr
          className="tableHeading"
        >
          <td>
            product
          </td>
          <td>
            quantity
          </td>
          <td>
            price
          </td>
        </tr>
        <tr
          className="white"
        >
          <td>
            chocolate cookies
          </td>
          <td
            className="quantityColumn"
          >
            1
          </td>
          <td
            className="confirmationPriceColumn"
          >
            € 
            2.39
          </td>
        </tr>
        <tr
          className="lightBlue"
        >
          <td
            colSpan={3}
            style={
              Object {
                "padding": "0 20px",
              }
            }
          >
            <hr />
          </td>
        </tr>
        <tr
          className="lightBlue"
        >
          <td
            className="sc-fzqARJ bQIqfM"
            colSpan={2}
          >
            <strong>
              surcharge:
            </strong>
          </td>
          <td
            className="sc-fzqNqU gRQPnX"
          >
            € 
            3.00
          </td>
        </tr>
        <tr
          className="lightBlue"
        >
          <td
            className="sc-fzoNJl gyueDb"
            colSpan={2}
          >
            <strong>
              total:
            </strong>
          </td>
          <td
            className="sc-fzoXWK jzJqLr"
          >
            € 
            5.39
          </td>
        </tr>
        <tr
          className="lightBlue"
        >
          <td
            colSpan={3}
            style={
              Object {
                "padding": "0 20px",
              }
            }
          >
            <hr />
          </td>
        </tr>
        <tr
          className="lightBlue"
        >
          <td
            className="sc-fznJRM bhObLI"
            colSpan={3}
          >
            <strong>
              date processed: 
            </strong>
            <span>
              1582123900751
            </span>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
  <div
    className="summaryColumn"
  >
    <table
      className="sc-fznWqX bzpSuQ detailsTable"
    >
      <tbody>
        <tr
          className="header"
        >
          <th
            colSpan={3}
          >
            delivery address
          </th>
        </tr>
        <tr>
          <td
            className="sc-fzoiQi joKFly lightBlue"
            colSpan={3}
          >
            joe bloggs
            <br />
            11111111
            <br />
            NY
            <br />
            <hr />
            joe@bloggs.com
            <br />
            11111111
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
